rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -206,30 +206,35 @@ contract RBTCWrapperProxy is ContractRegistryClient {
 
         ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
         ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+        uint256 totalSupplyBefore = _poolToken.totalSupply();
+        uint32 reserveRatio_ = _liquidityPoolConverter.reserveRatio();
+
         uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
         _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
         uint256 poolTokenAmount = _poolToken.balanceOf(address(this)).sub(poolTokenAmountBefore);
         
         success = _poolToken.transfer(msg.sender, poolTokenAmount);
         require(success, "Failed to transfer pool token to user");
 
-        (uint256 rsvBalance, , , , ) = _liquidityPoolConverter.reserves(wrbtcTokenAddress);
```

Comment:
I think instead of using a formula you can just record this contract's balances of `_reserveTokens` before and after `addLiquidity` and compare the difference with `_reserveAmounts`

Alternatively, you can check allowances to see if particular token has been fully utilized:
`uint unused = _reserveTokens[i].allowance(this, _liquidityPoolConverterAddress);` 
That's probably the cleanest solution so far.